spec.md

```diff
@@ -1090,6 +1088,223 @@ message ControllerServiceCapability {
 
 If the plugin is unable to complete the ControllerGetCapabilities call successfully, it MUST return a non-ok gRPC code in the gRPC status.
 
+#### `CreateSnapshot`
+
+A Controller Plugin MUST implement this RPC call if it has `CREATE_DELETE_SNAPSHOT` controller capability.
+This RPC will be called by the CO to create a new snapshot from a source volume on behalf of a user.
+
+This operation MUST be idempotent.
+If a snapshot corresponding to the specified snapshot `name` already exists and is compatible with the specified `source_volume_id` and `parameters` in the `CreateSnapshotRequest`, the Plugin MUST reply `0 OK` with the corresponding `CreateSnapshotResponse`.
```

Comment:
This is an important sentence: if the snapshot "exists" and the snapshot status is `ERROR` this RPC is still expected to return `OK`. Maybe we could make that more clear in the errors table that follows?

Reply:
If the snapshot "exists" and the snapshot status becomes ERROR, this RPC should be returning a gRPC error code, depending on what the error is.  Currently we have the following error codes specified: 6 ALREADY_EXISTS; 10 ABORTED (Operation pending for snapshot); 12 UNIMPLEMENTED; and 13 RESOURCE_EXHAUSTED.

Reply:
The sentence you are referring to only applies if the snapshot has been successfully cut and uploaded (if upload is part of the process).